src/Form/Form.js

```diff
@@ -18,6 +18,18 @@ function findFieldOption(options, field) {
   });
 }
 
+function mergeNewModel(stateModel, currentModelDefinition, nextModelDefinition) {
+  let newModel = {};
+
+  Object.keys(nextModelDefinition).forEach(function (key) {
+    if (currentModelDefinition[key] !== nextModelDefinition[key]) {
```

Comment:
yes, if the currentModelDefinition value is `null` and it isn't `null` anymore, then we want to replace it. it's not only for when a property is completely new